sympy/stats/crv.py

```diff
@@ -335,13 +335,24 @@ def where(self, condition):
         interval = interval.intersect(self.domain.set)
         return SingleContinuousDomain(rv.symbol, interval)
 
+
     def conditional_space(self, condition, normalize=True, **kwargs):
 
         condition = condition.xreplace(dict((rv, rv.symbol) for rv in self.values))
-
         domain = ConditionalContinuousDomain(self.domain, condition)
+
         if normalize:
-            pdf = self.pdf / domain.integrate(self.pdf, **kwargs)
+            # create a clone of the variable to
+            # make sure that variables in nested integrals are different
+            # from the variables outside the integral
+            # this makes sure that they are evaluated separately
+            # and in the correct order
+            replacement  = {rv: Dummy(str(rv)) for rv in self.symbols}
+            clone = self.xreplace(replacement)
+            clone_domain = ConditionalContinuousDomain(
+                            clone.domain, condition.xreplace(replacement))
+
+            pdf = self.pdf / clone_domain.integrate(clone.pdf, **kwargs)
```

Comment:
How does `clone_domain.integrate(clone.pdf, **kwargs)` differ from `domain.integrate(self.pdf, **kwargs).xreplace(replacement)`?

Reply:
You're right, it doesn't. Will fix this now. Thanks for mentoring me.